docs/spec_draft.md

```diff
@@ -846,6 +848,149 @@ defined and one of the following:
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.gather
+
+### Semantics
+
+Gathers slices from `operand` tensor from offsets specified in `start_indices`
+and produces a `result` tensor.
+
+The following diagram shows how elements in `result` map on elements in
+`operand` using a concrete example. The diagram picks a few example `result`
+indexes and explains in detail which `operand` indexes they correspond to.
+
+<img align="center" src="spec_images/gather.png" />
```

Comment:
Let's rename the directory to `spec_draft` to match the filename? #196 proposes a similar convention.